src/main/java/org/beanpod/switchboard/controller/EncoderController.java

```diff
@@ -31,24 +33,36 @@
 @RequestMapping("/encoder")
 @RequiredArgsConstructor
 public class EncoderController {
+
   public static final String UNKNOWN_ERROR_MESSAGE = "Unknown error in EncoderController";
 
   private final EncoderDaoImpl encoderDao;
   private final DeviceDaoImpl deviceService;
   private final EncoderMapper encoderMapper;
   private final StreamMapper streamMapper;
   private final EncoderService encoderService;
+  private final MaintainDeviceStatus maintainDeviceStatus;
 
   @GetMapping
   public List<EncoderDto> retrieveAllEncoders() {
     List<EncoderEntity> encoderEntities = encoderDao.getEncoders();
+    maintainDeviceStatus.maintainStatusField(encoderEntities);
     return encoderMapper.toEncoderDtos(encoderEntities);
   }
 
   @GetMapping("/{serialNumber}")
   public ResponseEntity<EncoderDto> retrieveEncoder(@PathVariable @Valid String serialNumber) {
-    return encoderDao
-        .findEncoder(serialNumber)
+
+    // maintain status field and create a log if status changed
+    Optional<EncoderDto> encoder = encoderDao.findEncoder(serialNumber);
+    if (encoder.isPresent()) {
+      List<EncoderEntity> encodersListTemp = new LinkedList<>();
+      encodersListTemp.add(encoderMapper.toEncoderEntity(encoder.get()));
+      maintainDeviceStatus.maintainStatusField(encodersListTemp);
+    }
+
+    // return encoder
+    return encoder
         .map(ResponseEntity::ok)
```

Comment:
returned encoder is not the maintained one i think

Reply:
Good catch! I've changed the code to return the most updated object.